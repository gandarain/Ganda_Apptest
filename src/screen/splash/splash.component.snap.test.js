import React from 'react'
import renderer from 'react-test-renderer'

import Splash from './splash.component'

jest
  .mock('@react-navigation/native', () => ({
    useNavigation: () => ({
      navigation: () => ({
        navigate: jest.fn()
      })
    })
  }))
  .mock('../../asset/image', () => ({
    images: () => ({
      logo: 'Logo'
    })
  }))

jest.useFakeTimers()

describe('Splash Component', () => {
  beforeEach(() => {
    let useEffect
    const mockUseEffect = () => {
      useEffect.mockImplementationOnce(f => f())
    }
    useEffect = jest.spyOn(React, 'useEffect')

    mockUseEffect()
  })

  it('should render splash component', () => {
    const rendered = renderer.create(<Splash />)
    const tree = rendered.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
