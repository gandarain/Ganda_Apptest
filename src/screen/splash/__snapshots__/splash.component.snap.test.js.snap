// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Splash Component should render splash component 1`] = `
<View
  accessibilityLabel="ContainerSplash"
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FF6049",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    accessibilityLabel="ImageSplash"
    style={
      {
        "height": 100,
        "width": 100,
      }
    }
  />
  <Text
    accessibilityLabel="TitleSplash"
    style={
      {
        "color": "#FFFFFF",
        "fontFamily": "Roboto-Bold",
        "fontSize": 20,
        "marginVertical": 20,
      }
    }
  >
    Contact
  </Text>
  <ActivityIndicator
    accessibilityLabel="ActivityIndicator"
    color="#FFFFFF"
    size="large"
  />
</View>
`;
